### PR TITLE
docs: Update deployment-modes.md

### LINKED ISSUE #9058: docs: add a link to the new one in the old stand-alone doc
### Current State

The doc for stand-alone mode is now:
https://apisix.apache.org/docs/apisix/deployment-modes/#standalone
Which is different from the path of the old one:
https://apisix.apache.org/docs/apisix/2.15/stand-alone/

Users may get confused about this:
https://github.com/apache/apisix/issues/9051

### Desired State

add a new link in the old stand-alone doc which point to the new one

### PR DESCRIPTION
**Pull request by ro4i7**
Added the new link to the old stand-alone doc #9058

### Description

Fixes #9058 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
It's the old stand-alone doc found first, can we add a new link to the old doc?
image

**Comment by ro4i7** [comment]
> It's the old stand-alone doc found first, can we add a new link in the old doc? image

@monkeyDluffy6017 yeah, Got it. I have updated it please check.

File: docs/en/latest/deployment-modes.md [hunk 1]
```diff
@@ -161,6 +161,8 @@ deployment:
 
 Turning on the APISIX node in Stand-alone mode will no longer use the default etcd as the configuration center.
 
+For the old version of the standalone mode deployment guide, see the [stand-alone mode documentation](https://apisix.apache.org/docs/apisix/deployment-modes/#standalone).
```
Comment (kayx23) [inline docs/en/latest/deployment-modes.md:164]: How is this the "old version"?

**Comment by kayx23** [comment]
The OG issue by @monkeyDluffy6017 states to add a new link in the old stand-alone doc to point to the new doc. This PR adds a link to itself and proposes changes for the master branch (which is the wrong branch). So it's not exactly what was asked.

@monkeyDluffy6017 - please check to see if the original req still persists. Thanks.

**Comment by monkeyDluffy6017** [comment]
We can close this pr and the related issue now, as we were able to find the right page during the search
image

**Pull request closed without merging**